modyn/trainer_server/internal/trainer/remote_downsamplers/remote_gradnorm_downsampling.py

```diff
@@ -31,30 +38,38 @@ def get_scores(self, forward_output: torch.Tensor, target: torch.Tensor) -> torc
             with torch.inference_mode():
                 # Because CrossEntropyLoss includes the softmax, we need to apply the
                 # softmax to the forward output to obtain the probabilities
-                probs = torch.nn.functional.softmax(forward_output, dim=1)
+                probs = F.softmax(forward_output, dim=1)
                 num_classes = forward_output.shape[-1]
-                one_hot_targets = torch.nn.functional.one_hot(target, num_classes=num_classes)
+                # pylint: disable=E1102
+                one_hot_targets = F.one_hot(target, num_classes=num_classes)
                 scores = torch.norm(probs - one_hot_targets, dim=-1)
         else:
             sample_losses = self.per_sample_loss_fct(forward_output, target)
             last_layer_gradients = torch.autograd.grad(sample_losses.sum(), forward_output, retain_graph=False)[0]
             scores = torch.norm(last_layer_gradients, dim=-1)
 
-        return scores
+        return scores.cpu()
```

Comment:
i am not sure I understand yet the score device handling. How exactly did you change that? In this PR, you change the device in the trainer, and here you add the .cpu() call, but I am not sure why

Reply:
If the device is the GPU, the scores are computed in GPU (the tensors are there, and pytorch automatically performs the operations in GPU). However, to select the samples, we have to keep all the scores, and keeping them all in GPU does not make sense. Therefore, the scores are calculated in GPU, moved to CPU memory and saved in a list.
If the device is the CPU, everything happens already in the CPU, and that line is useless.